frontend/src/components/charts/Indicator.js

```diff
@@ -7,6 +7,34 @@ import moment from 'moment';
 class Indicator extends Component {
   constructor(props) {
     super(props);
+    this.state = { localComputedTimestamp: null };
+  }
+
+  /**
+   * Updates the local computed timestamp with the one received from the props. Case when it comes from a query via xhr or via WS
+   * @param {props} param0
+   * @returns
+   */
+  static getDerivedStateFromProps({ data: { computedTimestamp } }) {
+    return { localComputedTimestamp: computedTimestamp };
+  }
+
+  /**
+   * @summary sets the local timestamp + 1 min, called on one min interval after the component mounts
+   */
+  updateTimestamp = () => {
```

Comment:
@siemiatj  do u have any suggestion how can we improve the clarify of this code because IMHO I find it  quite weird.

IMHO we shall not increment localComputedTimestamp but just set it when we get it from WS.

render() will just display the fuzzy time elapsed from `localComputedTimestamp` until `now`.

But in this case I think there is something more we have to do. We have to trigger somehow the component render. Not sure if we can do this programmatically....
Another idea would be to introduce some "counter" into the state and increment the counter from updateTimestamp() function.

I hope i made my concern clear.
@petrican  @siemiatj  wdyt?

Reply:
PS: also, current impl is very error phrone. at least the refresh interval (i.e. 1min) shall be extracted to a constant and used in updateTimestamp and componentDidMount.

but pls don't do this one. let's focus to get a better and cleaner code.

Reply:
@teosarca 

1) "IMHO we shall not increment localComputedTimestamp but just set it when we get it from WS." - the BE is not sending the data on every minute via WS. If it were to be sent on the FE side we should not need to have any incrementation of the computed timestamp
<img width="1585" alt="Screenshot 2021-06-29 at 07 06 44" src="https://user-images.githubusercontent.com/1708561/123736016-fca96f80-d8a8-11eb-9bf3-da89c78351cb.png">
Note: when the data is present via WS that localComputedTimestamp is also updated with the computedTimestamp within the data(from props).


2) "render() will just display the fuzzy time elapsed from localComputedTimestamp until now." - this is what it actually happens. I am using fromNow of the moment lib -> https://momentjs.com/docs/#/displaying/fromnow/ 
![Screenshot 2021-06-29 at 07 11 21](https://user-images.githubusercontent.com/1708561/123736135-46925580-d8a9-11eb-8f8e-cbe6a2e9e522.png). And the good part is already translated.

3) "But in this case I think there is something more we have to do. We have to trigger somehow the component render. Not sure if we can do this programmatically...." - The component state is updated at the end of the updateTimestamp function when localComputedTimestamp is updated. That change in the state is triggering the component to render after that update took place.

4) "Another idea would be to introduce some "counter" into the state and increment the counter from updateTimestamp() function." - this "counter" in my implementation is the actual timestamp which is incremented every minute with a min.
![Screenshot 2021-06-29 at 07 22 22](https://user-images.githubusercontent.com/1708561/123736960-cbca3a00-d8aa-11eb-84ea-4c33cb3a9352.png)

@siemiatj I am open for other ideas. Currently I don't have other better solutions that this. :(

Reply:
@petrican  pushed 2 commits to show you what I ment